packages/dapp/src/components/DualWidget/WidgetEvents.tsx

```diff
@@ -112,9 +121,19 @@ export function WidgetEvents() {
     );
     widgetEvents.on(WidgetEvent.RouteExecutionFailed, onRouteExecutionFailed);
     widgetEvents.on(WidgetEvent.RouteHighValueLoss, onRouteHighValueLoss);
+    widgetEvents.on(WidgetEvent.SafeRouteInitiated, onSafeRouteInitiation);
```

Comment:
We can reuse RouteExecutionStarted event here, no need to introduce a new one

Reply:
Got it, I'll pass an extra argument to indicate this is a SAFE Route

Reply:
I'd prefer a separate event:

while emitting the events, we can pass only 2 args in `emitter.emit()` and it's always `EventType` followed by `route/data`.
So, we won't be able to pass a simple distinguisher whether the route is SAFE route or not.

and currently, on the receiving end of these events, there is no logic present it's directly function's being called and execution.

Hence, following the same structure I made another event rather than introducing logic in Jumper for it.

Reply:
On Jumper we already have access to Safe SDK and we can determine if we are executing a route inside Safe Wallet. Also, we can subscribe to the same event in different places so it's easier to utilize it.

Reply:
moving the logic from widget to jumper and resorting to use the `RouteUpdatedEvent`